src/main/java/com/avioconsulting/mule/opentelemetry/internal/store/Transaction.java

```diff
@@ -7,12 +7,14 @@ public class Transaction implements Serializable {
   private final String rootFlowName;
   private final FlowSpan rootFlowSpan;
   private final String traceId;
+  private final String spanId;
 
   public Transaction(String transactionId, String traceId, String rootFlowName, FlowSpan rootFlowSpan) {
     this.transactionId = transactionId;
     this.rootFlowName = rootFlowName;
     this.rootFlowSpan = rootFlowSpan;
     this.traceId = traceId;
+    this.spanId = rootFlowSpan.getSpan().getSpanContext().getSpanId();
```

Comment:
A trace can include many spans (~ 1 per message processor) but rootFlowSpan will represent the root span associated with the trace id. I think we should rename it to `rootSpanId` so the intent is clear. 

Ideally, there will just be one root span per transaction in a request processing (except when bug #71)  so any logs tied to a trace id and span id should show logs for request processing per app.

Reply:
I was thinking it would just be the "current" span, as in the Span ID here not the Parent Span ID:
![image](https://github.com/avioconsulting/mule-opentelemetry-module/assets/5859975/45aa6a8f-cfba-40d1-b35e-99afd858be36)

If it would pick up the root one, and thats not the same as the above "Span ID", then this might not work as I thought it did. I'm not too knowledgable about the whole OTel structure as you might have guessed 😄

Reply:
The span id from `rootFlowSpan` will definitely be a root span id. The other child spans under the root will be created per processor, just prior to processor execution and ended right after it. I don't think you will be able to assign those to logs as you are expecting to.

We will plan to investigate the use of the lo4j context data library. Any internal logs from processors may be able to get span id but mule logger won't be able to assign log statements to other spans from other processors.